alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportAcknowledgeAlertAction.kt

```diff
@@ -94,7 +99,9 @@ class TransportAcknowledgeAlertAction @Inject constructor(
                 searchRequest,
                 object : ActionListener<SearchResponse> {
                     override fun onResponse(response: SearchResponse) {
-                        onSearchResponse(response)
+                        scope.launch {
+                            onSearchResponse(response)
+                        }
                     }
 
```

Comment:
you can get rid of `ActionListener` callback altogether. Just do `client.suspendUntil { search(searchRequest, it) }` and call `onSearchResponse()` after it in a direct style. You can make `findActiveAlerts` a suspend function.  Also, start the coroutine with defined scope in `doExecute()` to keep it clean.

Reply:
done